### PR TITLE
Cart Implementation Complete

### PR DESCRIPTION
**Pull request by NStro03**
Features implemented
Increment/Decrement item specific counts
Remove item from cart
Calculate and update Total Cost
Calculate and update Total Items in cart

### CODE REVIEW

File: src/context.js [hunk 1]
```diff
@@ -3,16 +3,46 @@ import cartItems from './data'
...
 // ATTENTION!!!!!!!!!!
 // I SWITCHED TO PERMANENT DOMAIN
-const url = 'https://course-api.com/react-useReducer-cart-project'
+// const url = 'https://course-api.com/react-useReducer-cart-project'
 const AppContext = React.createContext()
 
 const AppProvider = ({ children }) => {
-  const [cart, setCart] = useState(cartItems)
+  const totalItems = cartItems.reduce((count, item) => (count += item.amount), 0);
+  console.log("CountedItems: " + totalItems);
```
Comment (Ishaan-flyfin) [inline src/context.js]: remove logs

File: src/context.js [hunk 2]
```diff
@@ -3,16 +3,46 @@ import cartItems from './data'
...
+
+  const stateDefault = {
+    totalCost,
+    totalItems,
+    cart: cartItems
+  }
+  const [state, dispatch] = useReducer(reducer, stateDefault);
+
+  const incrementQtyForId = (id) => {
+    dispatch({ type: 'INCREMENT_ITEM', payload: id });
```
Comment (Ishaan-flyfin) [inline src/context.js]: we can keep the 'type' as constant and use the same in reducer as well

File: src/reducer.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        console.log('increase')
+        let newCost = state.totalCost;
+        let itemCount = state.totalItems;
+        let newCart = state.cart.map((item) => {
+            if (item.id === action.payload) {
+                itemCount += 1;
+                newCost += item.price;
+                return { ...item, amount: item.amount + 1 }
+            }
+            else { return item }
```
Comment (Ishaan-flyfin) [inline src/reducer.js]: here there is no use of `else` as we are returning in `if` 
we can simply do 
```
if() {
   return
} 

return
```

**Review by Ishaan-flyfin — CHANGES_REQUESTED** [review]

File: src/context.js (hunk 1 above)
Comment (NStro03) [reply src/context.js]: Done.

File: src/context.js (hunk 2 above)
Comment (NStro03) [reply src/context.js]: Moved all the "type"s to a constants.js file.

File: src/reducer.js (hunk 3 above)
Comment (NStro03) [reply src/reducer.js]: Done

**Pull request merged**